flit_core/flit_core/config.py

```diff
@@ -608,9 +608,13 @@ def read_pep621_metadata(proj, path) -> LoadedConfig:
                     raise ConfigError(
                         f"License file path ({license_f}) cannot be an absolute path"
                     )
+                if ".." in license_f:
```

Comment:
Perhaps? Although, `normpath` just does lexical manipulation and e.g. `a/spam/../b` might refer to a symlink.

```suggestion
                if ".." in os.path.normpath(license_f):
```

Reply:
Or actually, as e.g. a file might be named `a..b`. I think backslashes are prohibited?


```suggestion
                if ".." in license_f.split('/'):
```

Reply:
> Perhaps? Although, `normpath` just does lexical manipulation and e.g. `a/spam/../b` might refer to a symlink.

Technically, yes. Although I'm not sure it's worth the effort. Same for file names which contain `..`.
For the overwhelming majority, checking `if ".." in license_f:` to provide a better error message should be enough.

Reply:
Here's the check we do for paths in the sdist include/exclude lists:

https://github.com/pypa/flit/blob/5f8c75fba2f926036da6188ab8b59524ea4859f3/flit_core/flit_core/config.py#L239-L249

I can't think of a use case for a path like `foo/../bar` (which will normalise to just `bar`), but it is currently allowed in that case. Whatever we do, I'd like it to be consistent across all the places where a relative path is specified.